packages/core/src/utils.js

```diff
@@ -512,3 +512,19 @@ export class DefaultMap extends Map {
     return super.get(key);
   };
 };
+
+export function compareObjectTypes(obj1, obj2) {
+  if (obj1 === obj2) return true; // Handles primitives
```

Comment:
- Do we need this, as this only check obj1 and obj2 reference and not the actual value.
- Below we are already doing this check `key in key2`

Reply:
above is early exit for the primitive check as we don't want to spend time in converting data type and checking. Check lodash implementation may be.